src/convert/impls.rs

```diff
@@ -624,3 +625,123 @@ where
     }
     result.into_boxed_slice()
 }
+
+// WebAssembly ToWebAssemblyValue implementations for primitive types
+// These implement the exact WebAssembly specification algorithms
```

Comment:
Continuing from the other comment - @RReverser mentioned an alternative approach here is to import a JS function with various types to implicitly handle the casting without an explicit implementation.

This would involve defining `function (x) -> x` and then importing it as taking an externref argument, but returning as each of the 4 core Wasm types in its imported variations, then using that to handle casting implicitly.

My argument remains this is more code and work overall than just having these simple checks which are well-defined.